src/Bitcoin/DustCalculator.cpp

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright © 2017 Trust Wallet.
+
+#include "DustCalculator.h"
+
+namespace TW::Bitcoin {
+
+FixedDustCalculator::FixedDustCalculator(Amount fixed) noexcept
+    : fixedDustAmount(fixed) {
+}
+
+Amount FixedDustCalculator::dustAmount([[maybe_unused]] Amount byteFee) noexcept {
+    return fixedDustAmount;
+}
+
+LegacyDustCalculator::LegacyDustCalculator(TWCoinType coinType) noexcept
+    : feeCalculator(getFeeCalculator(coinType, false)) {
+}
+
+Amount LegacyDustCalculator::dustAmount([[maybe_unused]] Amount byteFee) noexcept {
+    return feeCalculator.calculateSingleInput(byteFee);
```

Comment:
Generally speaking, is it even appropriate to refer to this as `dustAmount`?

```cpp
int64_t LinearFeeCalculator::calculateSingleInput(int64_t byteFee) const noexcept {
    return static_cast<int64_t>(std::ceil(bytesPerInput)) * byteFee; // std::ceil(101.25) = 102
}
```

Reply:
No, but this is how our current chains work. This is a tradeoff to avoid breaking changes for other Bitcoin forks.